core/data_acquisition/websocket_consumers/queue_ws_consumer.py

```diff
@@ -0,0 +1,17 @@
+import multiprocessing
+from .websocket_consumer import WebsocketConsumer
+
+
+class QueueWSConsumer(WebsocketConsumer):
+    def __init__(self):
+        maxsize = 1024 # CHANGE THIS LATER ?? I don't know what this ideally should be
+        self.mp_queues = multiprocessing.Queue(maxsize)
+        pass
+
+    async def _async_consume(self, data_queue):
+        while True:
+            data = await data_queue.get()
+            self.mp_queues.put(data, True) # might need to change true to false later, idk
+            print("Put into queue: ", data)
+            # could be useful to call every time you add as a test - Queue.qsize() --> does not work on MacOS
```

Comment:
You can use something along the lines of isempty...

We are going to be running everything on Unix. When me and @jiale0402 ran into this like 11 months ago, we originally were going to implement in qsize but opted to give up on that and look for something universal. Regardless not a bad comment, just more context.